commons-component-common/src/main/java/org/exoplatform/services/thumbnail/ImageResizeServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.exoplatform.services.thumbnail;
+
+import org.imgscalr.Scalr;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class ImageResizeServiceImpl implements ImageResizeService {
+
+  @Override
+  public byte[] scaleImage(byte[] image, int width, int height, boolean fitExact, boolean ultraQuality) throws Exception {
+    if (width == 0 && height == 0) {
+      return image;
+    }
+    Scalr.Method resizeMethod = ultraQuality ? Scalr.Method.ULTRA_QUALITY : Scalr.Method.QUALITY;
+    BufferedImage bufferedImage = toBufferedImage(image);
+    if (width == 0) {
+      bufferedImage = Scalr.resize(bufferedImage, resizeMethod, Scalr.Mode.FIT_TO_HEIGHT, width, height, Scalr.OP_ANTIALIAS);
+    } else if (height == 0) {
+      bufferedImage = Scalr.resize(bufferedImage, resizeMethod, Scalr.Mode.FIT_TO_WIDTH, width, height, Scalr.OP_ANTIALIAS);
+    } else if (fitExact) {
+      bufferedImage = Scalr.resize(bufferedImage, resizeMethod, Scalr.Mode.FIT_EXACT, width, height, Scalr.OP_ANTIALIAS);
+    } else {
+      int originWidth = bufferedImage.getWidth();
+      int originHeight = bufferedImage.getHeight();
+      Scalr.Mode fitMode =
```

Comment:
Can you explain functionnaly this line ?

Reply:
> Can you explain functionnaly this line ?

Here i'm reversing the automatic fit of the Scalr because i want to have a better quality for images, specially for user avatars:
this will make sure to have a resized image with a width and height equal or superior to this target one, i mean when i want an image with 200x200, i want to have image in 200x260 for example when the image is not rectangular and not a 200x160 for example.

i hope i could explain :)

Reply:
In fact, if the boolean exactFit = false, we must keep the image ratio. It was already done in ImageUtils, and we must keep this behaviour.
So as I understand, if exactFit = false, then you take the larger between width and height, and then adapt the other one to keep the ratio. Is it right  ?

So I think this case should be described with a test case